### PR TITLE
More debug logging. on node state provider instantiation

### PR DESCRIPTION
**Pull request by patsonluk**
[CH187112]

## Description
From our last solr deployment on Feb 17, it's found that majority of the startup time is spent between line
https://github.com/fullstorydev/lucene-solr/blob/d7ed1d3bfbc038cf5c802f0b443d3d2b857bed3d/solr/core/src/java/org/apache/solr/cloud/ZkController.java#L366-L369

The time spent on each node can be found in
https://docs.google.com/spreadsheets/d/1NipJnUwEUrKbJ63Ld7v3N_sQgteXi40YAVtvPRPFJvA/edit?usp=sharing , which the longest one on c81 (c81-253) could take up to 6mins42sec

Upon further debugging, it's found that each node would try to fetch `state.json` for every single collection in the cluster:
1. During startup, ZkController `init()` fetches all collections from zkclient, path `/collections`, such list is stored as lazy collection states in `ZkStateReader`. key is collection name, the value of type ClusterState.CollectionRef is not loaded
2. In `SolrClientNodeStateProvider#readReplicaDetails` call getClusterState, which return the lazy collection states from `ZkStateReader`
3. Then the clusterState is iterated which trigger actual load for each ClusterState.CollectionRef, this calls `LazyCollectionRef#get` which invokes `ZkStateReader#getCollectionLive` which fetches collection state from zk with path <collection>/state.json 

However, we should first add more logging for `NodesSysPropsCacher/NodeStateProvider` initialization to confirm this is the main cause.

## Solution
Added logging `SolrClientNodeStateProvider`, especially during iteration of all collections/replica in `readReplicaDetails`. Since there are alot of collections per cluster, we will only print out message for roughly each 10% of collections processed.

### CODE REVIEW

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/SolrClientNodeStateProvider.java [hunk 1]
```diff
@@ -83,39 +84,55 @@
...
   }
 
   protected ClusterStateProvider getClusterStateProvider() {
     return solrClient.getClusterStateProvider();
   }
 
   protected void readReplicaDetails() throws IOException {
+    log.info("Started reading replica details");
     ClusterStateProvider clusterStateProvider = getClusterStateProvider();
+    log.info("Got clusterStateProvider " + clusterStateProvider.getClass().getName());
```
Comment (hiteshk25) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/SolrClientNodeStateProvider.java]: Maybe add explicit null check here?

**Review by hiteshk25 — APPROVED** [review]
LGTM (please make sure we test in playpen!)

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/SolrClientNodeStateProvider.java (hunk 1 above)
Comment (patsonluk) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/SolrClientNodeStateProvider.java]: Done! :)

**Comment by patsonluk** [comment]
Tested on playpen. Confirmed the new logs.

For example on playpen solros-c08-1:
```
2022-03-10 22:52:33.853 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Started processing collection states
2022-03-10 22:52:34.047 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 43 of 430 collection(s) for replica info
2022-03-10 22:52:34.097 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 86 of 430 collection(s) for replica info
2022-03-10 22:52:34.182 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 129 of 430 collection(s) for replica info
2022-03-10 22:52:34.223 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 172 of 430 collection(s) for replica info
2022-03-10 22:52:34.271 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 215 of 430 collection(s) for replica info
2022-03-10 22:52:34.320 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 258 of 430 collection(s) for replica info
2022-03-10 22:52:34.360 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 301 of 430 collection(s) for replica info
2022-03-10 22:52:34.402 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 344 of 430 collection(s) for replica info
2022-03-10 22:52:34.448 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 387 of 430 collection(s) for replica info
2022-03-10 22:52:34.490 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Processing 430 of 430 collection(s) for replica info
2022-03-10 22:52:34.491 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Finished reading replica details
2022-03-10 22:52:34.491 INFO  (main) [    ] o.a.s.c.s.i.SolrClientNodeStateProvider Created SolrClientNodeStateProvider
```

Same is observed on data node solr-c08-1.play.

While on solrqueryaggregator-c08-1, it does NOT iterate the replicas, which is expected:
```
2022-03-10 22:54:55.058 INFO  (main) [    ] o.a.s.c.ZkController initializing NodesSysPropsCacher
2022-03-10 22:54:55.060 INFO  (main) [    ] o.a.s.c.ZkController initialized NodesSysPropsCacher
```

**Pull request merged**